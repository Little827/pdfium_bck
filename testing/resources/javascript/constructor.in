{{header}}
{{object 1 0}} <<
  /Type /Catalog
  /Pages 2 0 R
  /OpenAction 10 0 R
>>
endobj
{{object 2 0}} <<
  /Type /Pages
  /Count 1
  /Kids [
    3 0 R
  ]
>>
endobj
% Page number 0.
{{object 3 0}} <<
  /Type /Page
  /Parent 2 0 R
  /MediaBox [0 0 612 792]
>>
% OpenAction action
{{object 10 0}} <<
  /Type /Action
  /S /JavaScript
  /JS 11 0 R
>>
endobj
% JS program to execute
{{object 11 0}} <<
  {{streamlen}}
>>
stream
{{include constructor.js}}
testIllegalConstructor("this");
testIllegalConstructor("globalThis");
testIllegalConstructor("app");
testIllegalConstructor("event");
testIllegalConstructor("font");
testIllegalConstructor("global");
testIllegalConstructor("util");
testIllegalConstructor("style");
testIllegalConstructor("color");
testIllegalConstructor("border");
testIllegalConstructor("display");
testIllegalConstructor("console");
testIllegalConstructor("position");
testIllegalConstructor("highlight");
testIllegalConstructor("zoomtype");
testIllegalConstructor("scaleHow");
testIllegalConstructor("scaleWhen");
endstream
endobj
{{xref}}
{{trailer}}
{{startxref}}
%%EOF
