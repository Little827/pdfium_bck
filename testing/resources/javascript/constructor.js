// Copyright 2020 PDFium Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

function testIllegalConstructor(name, allowed) {
  const constructorString = name + ".constructor";
  let constructor;
  try {
    constructor = eval(constructorString);
  } catch (e) {
    app.alert("FAIL: No such " + constructorString);
    return;
  }
  try {
    constructor();
    app.alert("FAIL: " + constructorString + "(): returned");
  } catch (e) {
    app.alert("PASS: " + constructorString + "(): " + e);
  }
  try {
    new constructor;
    app.alert("FAIL: new " + constructorString + ": returned");
  } catch (e) {
    app.alert("PASS: new " + constructorString + ": " + e);
  }
}
