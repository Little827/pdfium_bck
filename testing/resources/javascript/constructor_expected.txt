Alert: PASS: this.constructor(): Illegal Constructor
Alert: PASS: new this.constructor: Illegal Constructor
Alert: PASS: app.constructor(): Illegal Constructor
Alert: PASS: new app.constructor: Illegal Constructor
Alert: PASS: border.constructor(): Illegal Constructor
Alert: PASS: new border.constructor: Illegal Constructor
Alert: PASS: color.constructor(): Illegal Constructor
Alert: PASS: new color.constructor: Illegal Constructor
Alert: PASS: console.constructor(): Illegal Constructor
Alert: PASS: new console.constructor: Illegal Constructor
Alert: PASS: display.constructor(): Illegal Constructor
Alert: PASS: new display.constructor: Illegal Constructor
Alert: PASS: event.constructor(): Illegal Constructor
Alert: PASS: new event.constructor: Illegal Constructor
Alert: PASS: font.constructor(): Illegal Constructor
Alert: PASS: new font.constructor: Illegal Constructor
Alert: PASS: global.constructor(): Illegal Constructor
Alert: PASS: new global.constructor: Illegal Constructor
Alert: PASS: highlight.constructor(): Illegal Constructor
Alert: PASS: new highlight.constructor: Illegal Constructor
Alert: PASS: position.constructor(): Illegal Constructor
Alert: PASS: new position.constructor: Illegal Constructor
Alert: PASS: scaleHow.constructor(): Illegal Constructor
Alert: PASS: new scaleHow.constructor: Illegal Constructor
Alert: PASS: scaleWhen.constructor(): Illegal Constructor
Alert: PASS: new scaleWhen.constructor: Illegal Constructor
Alert: PASS: style.constructor(): Illegal Constructor
Alert: PASS: new style.constructor: Illegal Constructor
Alert: PASS: util.constructor(): Illegal Constructor
Alert: PASS: new util.constructor: Illegal Constructor
Alert: PASS: zoomtype.constructor(): Illegal Constructor
Alert: PASS: new zoomtype.constructor: Illegal Constructor
